### PR TITLE
Add test_replica_failure_during_attaching test

### PR DESCRIPTION
**Pull request by meldafrawi**
https://github.com/longhorn/longhorn/issues/1999

Author: khushboo-rancher <60111667+khushboo-rancher@users.noreply.github.com
Co-authored-by: Mohamed Eldafrawi <mohamed.eldafrawi@rancher.com>
Signed-off-by: Mohamed Eldafrawi <mohamed.eldafrawi@rancher.com>

### CODE REVIEW

File: manager/integration/tests/test_ha.py [hunk 1]
```diff
@@ -2370,4 +2359,68 @@ def test_replica_failure_during_attaching():
...
+    volume_2 = client.by_id_volume(volume_name_2)
+    write_volume_random_data(volume_2)
+
+    del update_disks["extra-disk"]
+    node.diskUpdate(disks=update_disks)
+    common.wait_for_disk_update(client, node.name, 1)
+
+    replenish_wait_setting = \
+        client.by_id_setting(SETTING_REPLICA_REPLENISHMENT_WAIT_INTERVAL)
+    client.update(replenish_wait_setting, value='600')
```
Comment (shuo-wu) [inline manager/integration/tests/test_ha.py:2426]: nodes and settings will be reset automatically

**Review by shuo-wu — CHANGES_REQUESTED** [review]
This step is missed. And this step result in the skeleton is wrong. Can you update it to `--> Verify a new replica will be created but it cannot be scheduled.`?
```
10. Wait for the replenishment interval.
        --> Verify the failed replica using the extra disk will be removed.
```

File: manager/integration/tests/test_ha.py (hunk 1 above)
Comment (meldafrawi) [reply manager/integration/tests/test_ha.py]: fixed

File: manager/integration/tests/test_ha.py [hunk 2]
```diff
@@ -2365,9 +2354,76 @@ def test_replica_failure_during_attaching():
     9. Attach volume2.
        --> Verify volume will be attached with state Degraded.
     10. Wait for the replenishment interval.
-        --> Verify the failed replica using the extra disk will be removed.
+        ## --> Verify the failed replica using the extra disk will be removed.
```
Comment (shuo-wu) [inline manager/integration/tests/test_ha.py:2357]: You can directly remove it... 
In general, LGTM

Comment (meldafrawi) [reply manager/integration/tests/test_ha.py]: fixed

**Review by shuo-wu — APPROVED** [review]

**Pull request merged**